src/+Models/CaffeModel.m

```diff
@@ -0,0 +1,144 @@
+        has_thr;
+    end
+    
+    properties (SetAccess = ?ModelTrainers.CaffeModel, Transient=false)
+        net;
+    end
+    
+    %% --------------------------------------------------------------------
+    methods
+
+        function obj = CaffeModel(dir_matcaffe, ...
+                modelDir, fname_net_def, fname_weights, ...
+                thr)
+           % Add caffe/matlab to our Matlab search PATH to use matcaffe
+            if exist([dir_matcaffe, filesep, '+caffe'], 'dir')
+                addpath(dir_matcaffe);
+            else
+                error('Could not find matcaffe.');
+            end
+            obj.net = [];
+            obj.initNet(modelDir, fname_net_def, fname_weights);
+            if exist('thr', 'var') && ~isempty(thr)
+                obj.thr = thr;
+                if ~isa(obj.thr, 'cell')
+                    obj.thr = {obj.thr};
+                end
+                obj.has_thr = true;
+            else
+                obj.has_thr = false;
+            end
+        end
+        %% -----------------------------------------------------------------
+        
+
+        %% -----------------------------------------------------------------
+        function initNet(obj, modelDir, fname_net_def, fname_weights)
+            % INITNET
+            % innitialize underlying caffe network object from definition
+            % and weight files
+            %% --------------------------------------------------------------------
+            
+%             if ~exist(modelDir, 'dir')
+%                 error('Could not find model directory %s.', modelDir);
+%             end
+%             if ~exist(fullfile(modelDir, fname_net_def), 'file')
+%                 error('Could not find network definition prototxt.');
+%             else
+%                 obj.fpath_net_def = fullfile(modelDir, fname_net_def);
+%             end
+%             if ~exist(fullfile(modelDir, fname_weights), 'file')
+%                 error('Could not find network weights file.');
+%             else
+%                 obj.fpath_weights = fullfile(modelDir, fname_weights);
+%             end
+
+            obj.fpath_net_def = xml.dbGetFile(fullfile(modelDir, fname_net_def));
+            obj.fpath_weights = xml.dbGetFile(fullfile(modelDir, fname_weights));
+            
+            phase = 'test'; % run with phase test (so that dropout isn't applied)
+            if ~isempty(obj.net)
+                delete(obj.net);
+                clear obj.net;
+            end
+            obj.net = caffe.Net(obj.fpath_net_def, obj.fpath_weights, phase);
+        end
+        %% -----------------------------------------------------------------
+        
+    end
+    
+    %% -----------------------------------------------------------------------------------
+    methods(Access = protected)
+        
+        function [y,score] = applyModelMasked( obj, x )
+            blobs_in = x{1};

```

Comment:
true, if featureMask in super class is not specified, x is handed over as is. In this case the KS calling applyModel( x ) defines x as a cell array, where each element is a feature matrix.
The network needs the individual features (ratemaps, ams...) as separate matrices, and the KS performs this separation.
this wouldn't work if featureMask defined in Model.Base is not empty.